### PR TITLE
Added identity commands + --assign-identity flag to containerapp create

### PR DESCRIPTION
**Pull request by haroonf**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/containerapp/azext_containerapp/custom.py [hunk 1]
```diff
@@ -124,6 +125,28 @@ def create_containerapp(cmd,
...
+    elif assign_user_identities: 
+        identity_def["type"] = "UserAssigned"
+
+    if assign_user_identities:
+        identity_def["userAssignedIdentities"] = {}
+        subscription_id = get_subscription_id(cmd.cli_ctx)
+        
+        for r in assign_user_identities:
+            r = _ensure_identity_resource_id(subscription_id, resource_group_name, r)
+            identity_def["userAssignedIdentities"][r] = {}
```
Comment (calvinsID) [inline src/containerapp/azext_containerapp/custom.py:148]: Is `principalId` and `clientId` for a user assigned identity populated by the api?

File: src/containerapp/azext_containerapp/custom.py [hunk 2]
```diff
@@ -124,6 +125,28 @@ def create_containerapp(cmd,
     config_def["ingress"] = ingress_def
     config_def["registries"] = [registries_def] if registries_def is not None else None
 
+    # Identity actions
```
Comment (calvinsID) [inline src/containerapp/azext_containerapp/custom.py:128]: Nit: wrap this in an if statement: If assign_identity is an empty array, we can just set identity_def = None

File: src/containerapp/azext_containerapp/custom.py [hunk 3]
```diff
@@ -561,3 +585,167 @@ def delete_managed_environment(cmd, name, resource_group_name, no_wait=False):
...
+    # Assign correct type
+    try:
+        if containerapp_def["identity"]["type"] != "None": 
+            if containerapp_def["identity"]["type"] == "SystemAssigned" and assign_user_identities:
+                containerapp_def["identity"]["type"] = "SystemAssigned,UserAssigned"
+            if containerapp_def["identity"]["type"] == "UserAssigned" and assign_system_identity:
+                containerapp_def["identity"]["type"] = "SystemAssigned,UserAssigned"
+        else: 
+            if assign_system_identity and assign_user_identities:
+                containerapp_def["identity"]["type"] = "SystemAssigned, UserAssigned"
```
Comment (calvinsID) [inline src/containerapp/azext_containerapp/custom.py:632]: Should the space be removed between "SystemAssigned" and "UserAssigned"?

File: src/containerapp/azext_containerapp/custom.py [hunk 4]
```diff
@@ -561,3 +585,167 @@ def delete_managed_environment(cmd, name, resource_group_name, no_wait=False):
...
+    except:
+        containerapp_def["identity"] = {}
+        containerapp_def["identity"]["type"] = "None"
+
+    if assign_system_identity and containerapp_def["identity"]["type"].__contains__("SystemAssigned"):
+        logger.warning("System identity is already assigned to containerapp")
+
+    # Assign correct type
+    try:
+        if containerapp_def["identity"]["type"] != "None":
```
Comment (calvinsID) [inline src/containerapp/azext_containerapp/custom.py:625]: Do we need capitalization checks here?

File: src/containerapp/azext_containerapp/custom.py [hunk 5]
```diff
@@ -561,3 +585,167 @@ def delete_managed_environment(cmd, name, resource_group_name, no_wait=False):
...
+        r = ContainerAppClient.show(cmd=cmd, resource_group_name=resource_group_name, name=name)
+    except CLIError as e:
+        handle_raw_exception(e)
+
+    try:
+        return r["identity"]
+    except: 
+        r["identity"] = {}
+        r["identity"]["type"] = "None"
+        return r["identity"]
```
Comment (calvinsID) [inline src/containerapp/azext_containerapp/custom.py:749]: Should we return string "None", None or [] if there are no identities?

File: src/containerapp/azext_containerapp/custom.py (hunk 1 above)
Comment (haroonf) [reply src/containerapp/azext_containerapp/custom.py:148]: Yes

File: src/containerapp/azext_containerapp/custom.py [hunk 6]
```diff
@@ -561,3 +585,167 @@ def delete_managed_environment(cmd, name, resource_group_name, no_wait=False):
...
+
+    except Exception as e:
+        handle_raw_exception(e)
+
+    
+def remove_managed_identity(cmd, name, resource_group_name, identities, no_wait=False):
+    _validate_subscription_registered(cmd, "Microsoft.App")
+
+    remove_system_identity = '[system]' in identities
```
Comment (calvinsID) [inline src/containerapp/azext_containerapp/custom.py:670]: Do we need capitalization checks?

File: src/containerapp/azext_containerapp/custom.py [hunk 7]
```diff
@@ -561,3 +585,167 @@ def delete_managed_environment(cmd, name, resource_group_name, no_wait=False):
...
+        containerapp_def["identity"]["type"]
+    except:
+        containerapp_def["identity"] = {}
+        containerapp_def["identity"]["type"] = "None"
+
+    if containerapp_def["identity"]["type"] == "None":
+        raise CLIError("The containerapp {} has no system or user assigned identities.".format(name))
+
+    if remove_system_identity:
+        if containerapp_def["identity"]["type"] == "UserAssigned":
```
Comment (calvinsID) [inline src/containerapp/azext_containerapp/custom.py:701]: What if type is "SystemAssigned,UserAssigned"?

File: src/containerapp/azext_containerapp/custom.py [hunk 8]
```diff
@@ -561,3 +585,167 @@ def delete_managed_environment(cmd, name, resource_group_name, no_wait=False):
...
+            raise CLIError("The containerapp {} has no system assigned identities.".format(name))
```
Comment (calvinsID) [inline src/containerapp/azext_containerapp/custom.py:702]: Design decision: Should we raise error when removing a non existing identity, or ignore and continue?

File: src/containerapp/azext_containerapp/custom.py [hunk 9]
```diff
@@ -561,3 +585,167 @@ def delete_managed_environment(cmd, name, resource_group_name, no_wait=False):
...
+    _validate_subscription_registered(cmd, "Microsoft.App")
+
+    remove_system_identity = '[system]' in identities
+    remove_user_identities = [x for x in identities if x != '[system]']
+    remove_id_size = len(remove_user_identities)
+
+    # Remove duplicate identities that are passed and notify
+    remove_user_identities = list(set(remove_user_identities))
+    if remove_id_size != len(remove_user_identities):
+        logger.warning("At least one identity was passed twice.")
```
Comment (calvinsID) [inline src/containerapp/azext_containerapp/custom.py:677]: User could also have passed in additional identities that are not currently on the containerapp. Should we just ignore if additional identities are passed that are not currently on the containerapp?

File: src/containerapp/azext_containerapp/custom.py [hunk 10]
```diff
@@ -561,3 +585,167 @@ def delete_managed_environment(cmd, name, resource_group_name, no_wait=False):
...
+                    containerapp_def["identity"]["userAssignedIdentities"].pop(old_user_identity)
+                    wasRemoved = True
+                    break
+
+            if not wasRemoved:
+                raise CLIError("The containerapp does not have specified user identity '{}' assigned, so it cannot be removed.".format(given_id))
+
+        if containerapp_def["identity"]["userAssignedIdentities"] == {}:
+            containerapp_def["identity"]["userAssignedIdentities"] = None
+            containerapp_def["identity"]["type"] = ("None" if containerapp_def["identity"]["type"] == "UserAssigned" else "SystemAssigned")
```
Comment (calvinsID) [inline src/containerapp/azext_containerapp/custom.py:727]: Could type be "SystemAssigned,UserAssigned" here?

File: src/containerapp/azext_containerapp/custom.py [hunk 11]
```diff
@@ -561,3 +585,167 @@ def delete_managed_environment(cmd, name, resource_group_name, no_wait=False):
...
+    
+    try:
+        r = ContainerAppClient.create_or_update(cmd=cmd, resource_group_name=resource_group_name, name=name, container_app_envelope=containerapp_def, no_wait=no_wait)
```
Comment (calvinsID) [inline src/containerapp/azext_containerapp/custom.py:730]: Need to do `_get_existing_secrets` here as well

File: src/containerapp/azext_containerapp/custom.py (hunk 3 above)
Comment (haroonf) [reply src/containerapp/azext_containerapp/custom.py:632]: Works both ways, server returns with a space so figured I'd follow suit

File: src/containerapp/azext_containerapp/custom.py (hunk 5 above)
Comment (haroonf) [reply src/containerapp/azext_containerapp/custom.py:749]: Legacy apps do not have identity param returned, so to keep consistency between legacy and new CAs I assign the server returned value manually.

**Pull request merged**